### DISCUSSION TITLE
A multiple-purpose Linter or multiple single-purpose linters ?

### DISCUSSION THREAD

**Discussion by alingse** [golangci/golangci-lint#4865] [Q&A]
Which one is recommend ?

I have written a linter with two types of lint logic, and I'm considering adding more. I'm not sure if this is a good approach compared to the style used by golangci-lint.

the linter is https://github.com/alingse/sundrylint/issues/1

**Comment by ldez** [comment]
It's better not to have multiple unrelated purposes inside a linter.

Also, it's important to search if a linter already handles the topic, if that's the case it's better to add rules inside this linter.

**Reply by alingse** [reply]
🤔 I search the exists linters, Maybe should I add the `rangeappendall` check rule to https://github.com/go-critic/go-critic/tree/master/checkers ? How do you think? is it related?
